File: 2161. Partition Array According to Given Pivot.py

# Link - https://leetcode.com/problems/partition-array-according-to-given-pivot/

# Space: O(n)
# Time: O(n)

class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        
        cur, answer = 0, [0]*len(nums)
        
        for op in [pivot.__gt__, pivot.__eq__, pivot.__lt__]:
            for i in range(len(nums)):
                if op(nums[i]):
                    answer[cur] = nums[i]
                    cur += 1
                    
        return answer

File: Leetcode/2161. Partition Array According to Given Pivot.cpp
/*
    https://leetcode.com/problems/partition-array-according-to-given-pivot/
    Time: O(n)
    Space: O(n)
*/

class Solution {
public:
    vector<int> pivotArray(vector<int> &nums, int pivot) {

        vector<int> res;

        for (auto x : nums)
            if (x < pivot)
                res.push_back(x);

        for (auto x : nums)
            if (x == pivot)
                res.push_back(x);

        for (auto x : nums)
            if (x > pivot)
                res.push_back(x);

        return res;
    }
};